marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/hub-entities.xqy

```diff
@@ -590,9 +590,9 @@ declare %private function hent:add-indexes-for-entity-properties($entities as js
                   fn:not(fn:starts-with(map:get($items, "$ref"), "#")) and
                   map:get($entity-type-properties, $entity-type-property)=>map:get("sortable")
             let $_ :=
-                let $title-property := fn:concat($entity-title, ".", $entity-type-property)
+                let $title-property-path := fn:concat("//es:instance", "/", $entity-title, "/", $entity-type-property)
                 where $is-sortable and fn:not($is-facetable)
-                return map:put($sortable-properties, $title-property, "")
+                return map:put($sortable-properties, $title-property-path, "")
```

Comment:
We have by default facet values set to true in the search options. So it returns search options for all indexes unless the facet=false is set for the range constraint.
As per DHFPROD-5017 and DHFPROD-5018, facets should be enabled only for facetable properties and indexes from structured entityType properties and disabled for sortable properties.
So the map contains the values that are sortable and not facetable, so that when a node is reached in the fix-options and exists in the map, the facet=false attribute is set on the range constraint,

Reply:
I believe this statement from 5017 - "If a property has been marked as "sortable" but not "facetable", then the user should be able to sort on the property but not be able to use it as a facet." - only means that it's possible for a property to be sortable but not facetable. But it's not required that a property be not facetable in order for it to be facetable. That is - if a property is sortable=true and facetable=true, we should see both a facet for it and the ability to sort it.

Reply:
Yes. The following happens now. There are tests included for this scenario. The title entityProperty is both sortable and facetable and facet=true attribute is set on the range constraint and it will also have the ability to sort. 

> That is - if a property is sortable=true and facetable=true, we should see both a facet for it and the ability to sort it.